include/scrimmage/simcontrol/SimControl.h

```diff
@@ -64,64 +64,215 @@ enum class EndConditionFlags {TIME = 1, ONE_TEAM = 2, NONE = 3, ALL_DEAD = 4};
 
 class SimControl {
  public:
+    /**
+     * @brief SimControl default constructor
+     */
     SimControl();
-    bool init();
-    void start();
-    void display_progress(bool enable);
+
+    /**
+     * @brief Initialize a scrimmage simulation.
+     *
+     * @param [in] mission_file The scrimmage mission file for the
+     * simulation. The mission_file can be a relative path, absolute path, or a
+     * scrimmage mission file that is already on the SCRIMMAGE_MISSION_PATH.
+     *
+     * When using SimControl to run a scrimmage mission, this init() function
+     * must always be called with a mission_file to properly initialize the
+     * simulation. After calling init(), the API supports running scrimmage in
+     * its own thread (e.g., using the run_threaded() function) or the
+     * controlling program can step the simulation manually (e.g., using the
+     * run_single_step() function).
+     */
+    bool init(const std::string& mission_file);
+
+    /**
+     * @brief Starts the mission by generating entities and setting up logging.
+     *
+     * Initializes entities, sets up logging, initializes the time.
+     */
+    bool start();
+
+    /**
+     * @brief Runs the scrimmage simulation in the current thread.
+     *
+     * Example API usage:
+     *
+     *  \code{.cpp}
+     *  SimControl simcontrol;
+     *  simcontrol.init("./missions/straight.xml");
```

Comment:
should `start` be mentioned in this comment? It seems like a tutorial is a better place for this documentation since users won't typically know to look here.

Reply:
I'll add this to the tutorials. In this case, start() doesn't have be called because it is called automatically by run(). When using the run_single_step() function directly, the user has to call start() explicitly. My motivation for allowing these different processes is to make unit testing easier. Using the run_single_step() in a unit test allows the tester to check the state of the simulation at each time step without having to wait for the entire simulation to complete. This also allows the testing unit to inject published messages at specific simulation times.